### PR TITLE
refactor: change static merkle to trie merkle

### PR DESCRIPTION
**Pull request by KaoImin**
**What this PR does / why we need it**:

This PR changes the calculation of transactions root and receipts root from static merkle to trie merkle.

**Which issue(s) this PR fixes**:

Fixes #

**Which docs this PR relation**:

Ref #

**Which toolchain this PR adaption**:

No Breaking Change

**Special notes for your reviewer**:

NIL

**CI Description**

| CI Name                                   | Description                                                     |
| ----------------------------------------- | --------------------------------------------------------------- |
| *Chaos CI*                                | Test the liveness and robustness of Axon under terrible network condition    |
| *Cargo Clippy*                            | Run `cargo clippy --all --all-targets --all-features`      |
| *Coverage Test*                           | Get the unit test coverage report                             |
| *E2E Test*                                | Run end-to-end test to check interfaces                         |
| *Code Format*                             | Run `cargo +nightly fmt --all -- --check` and `cargo sort -gwc`     |
| *Web3 Compatible Test*                    | Test the Web3 compatibility of Axon                               |
| *v3 Core Test*                            | Run the compatibility tests provided by Uniswap V3             |
| *OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19* | Run the compatibility tests provided by OpenZeppelin           |

**CI Usage**

> Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [x] Chaos CI
- [x] Cargo Clippy
- [x] Coverage Test
- [x] E2E Tests
- [x] Code Format
- [x] Unit Tests
- [x] Web3 Compatible Tests
- [x] OCT 1-5 And 12-15
- [x] OCT 6-10
- [x] OCT 11
- [x] OCT 16-19
- [x] v3 Core Tests

### CODE REVIEW

File: common/merkle/Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,9 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+cita_trie = "3.0"
+hasher = "0.1"
+rlp = "0.5"
 static_merkle_tree = "1.1.0"
```
Comment (hongda3141) [inline common/merkle/Cargo.toml:12]:
```suggestion
static_merkle_tree = "1.1"
```

**Review by hongda3141 — APPROVED** [review]

File: common/merkle/src/lib.rs [hunk 2]
```diff
@@ -47,3 +54,43 @@ fn merge(left: &Hash, right: &Hash) -> Hash {
     root.extend_from_slice(right);
     Hasher::digest(Bytes::from(root))
 }
+
+pub struct TrieMerkle(PatriciaTrie<MemoryDB, HasherKeccak>);
+
+impl Default for TrieMerkle {
+    fn default() -> Self {
+        TrieMerkle(PatriciaTrie::new(
+            Arc::new(MemoryDB::new(false)),
```
Comment (driftluo) [inline common/merkle/src/lib.rs:63]: why it must be arc here?

Comment (KaoImin) [reply common/merkle/src/lib.rs:63]: This is required by the `cita_trie` crate. https://docs.rs/cita_trie/latest/cita_trie/struct.PatriciaTrie.html#method.new

**Review by driftluo — APPROVED** [review]

File: common/merkle/Cargo.toml (hunk 1 above)
Comment (KaoImin) [reply common/merkle/Cargo.toml:12]: fixed in #869

**Pull request merged**